sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/ItemComparator.java

```diff
@@ -41,6 +44,17 @@ public int compare(Object obj1, Object obj2) {
             return Double.compare(((Number) obj1).doubleValue(), ((Number) obj2).doubleValue());
         case String:
             return ((String) obj1).compareTo((String) obj2);
+        case ArrayNode:
+        case ObjectNode:
+            try{
+                UInt128 hash1 = DistinctHash.getHash(obj1);
+                UInt128 hash2 = DistinctHash.getHash(obj2);
+                return hash1.compareTo(hash2);
+            }
+            catch (IOException e) {
+                throw new IllegalStateException(String.format("Exception for getting hash for type %s: %s ", type1.toString(), e.getMessage()));
```

Comment:
with this you will be loosing the cause exception stacktrace.

Reply:
I see, good point, changed to following: 
new IllegalStateException(String.format("Getting hash exception for type %s ", type1.toString()), e)